### PR TITLE
On Masked path, check if /dev/null already mounted before mounting

### PR DESCRIPTION
**Pull request by rhatdan**
We do not want to mount /dev/null over a masked path, if the path is
already /dev/null.

This prevents an containers running buildah from requiring additional privs
to mount on a /dev/null, when the target is already mounted.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@nalind @giuseppe @vrothberg @TomSweeneyRedHat PTAL

This fixes the issue we are seeing with people trying to run buildah within a locked down container with chroot.  SELinux is blocking the mounting of the second /dev/null on /proc/kcore.  /proc/kcore already has a /dev/null mounted on it.

File: chroot/run.go [hunk 1]
```diff
@@ -1355,8 +1362,8 @@ func setupChrootBindMounts(spec *specs.Spec, bundlePath string) (undoBinds func(
 				}
 			}
 		} else {
-			// The target's not a directory, so bind mount os.DevNull over it, unless it's already os.DevNull.
-			if !os.SameFile(nullinfo, targetinfo) {
+			// If the target's is not a directory or os.DevNull, bind mount os.DevNull over it.
```
Comment (TomSweeneyRedHat) [inline chroot/run.go]: nit of a nit, 
s/target's/target/
But only if you've other changes.

**Comment by TomSweeneyRedHat** [comment]
Changes LGTM, test issues don't look to be related.

**Comment by rhatdan** [comment]
@vrothberg Not sure how I could write a test to test this.  I have tested this by hand though.

@rh-atomic-bot r=TomSweeneyRedHat

**Pull request closed without merging**